src/components/menu/menu.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { FC, MutableRefObject, PropsWithChildren } from 'react'
+import { useClickOutside } from '../../hooks/use-click-outside'
+import { MenuItemGroup } from './menu.atoms'
+import { MenuProvider, useMenu } from './provider'
+import { ElMenu, ElMenuItemContainer, ElMenuPopover } from './styles'
+
+const Trigger = ({ children }) => {
+  const { triggerProps, isOpen } = useMenu()
+
+  return children({ triggerProps, isOpen })
+}
+
+const Popover: FC<PropsWithChildren<{ containerRef: MutableRefObject<HTMLDivElement | null> }>> = ({
+  children,
+  containerRef,
```

Comment:
**question:** why is this `containerRef` supplied by the consumer? I would have assumed the consumer doesn't need to concern itself with this and this component would simply set it up internally and attach it to the `ElMenuPopover` rendered here 🤔

Reply:
its because it used in `useClickOutside` hook, the hook basically does two thing:
1. register `handleClick` event listener to the 'document`
2. `handleClick` then will check if clicked `target` is outside the `containerRef` it will run the passed callback ( in this case is "closing menu")

Reply:
> its because it used in `useClickOutside` hook

Yes, but couldn't this component take care of the ref internally?

```javascript
const Popover: FC<...> = ({ children }) => {
  const { setIsOpen, ... } = useMenu()
  const ref = useRef<HTMLDivElement>(null);

  useClickOutside(ref, ...)

  if (isOpen) {
    // ...
    return (
      <ElMenuPopover ref={ref} ...>
        {children}
      </ElMenuPopover>
    )
  }

  return null
}
```

Reply:
> Yes, but couldn't this component take care of the ref internally instead of relying on a consumer to set it up?

I think no, its because otherwise the callback ("closeMenu") will trigger when user clicking the button triggerer

Reply:
I think you're correct, the button triggerer what I mean't here is the "Portfolio: All Portfolios" as the gif shown which is what I expecting, but what I can think of how it does is make the popover as an overlay then we don't have to worry about button triggerer to be clicked too, but now the question when menu is open (and overlay is covering evertyhing) it could also hide `:hover` style of other element behind, is that correct or it was totally different logic ?

Reply:
If I'm understanding you correctly, I don't think the popover element should be covering everything in the browser's viewport (this would prevent the popover from being able to accurately reposition itself when there's not enough space above/below the trigger button). Also, I'm not aware of any design requirement for the popover to have a backdrop like our dialogs and drawers do.

Does that answer your question? If not, we can scheduled a call next week to discuss verbally 👍

Reply:
yes you right, if console team doesn't use overlay than it should be different logic, i will try that 👍